Tests/Marketplace/search_and_install_packs.py

```diff
@@ -480,32 +481,46 @@ def install_all_content_packs_for_nightly(client: demisto_client, host: str, ser
     install_packs(client, host, all_packs, is_nightly=True)
 
 
-def install_all_content_packs(client: demisto_client, host: str, server_version: str):
-    """ Iterates over the packs currently located in the Packs directory. Wrapper for install_packs.
-    Retrieving the latest version of each pack from the metadata file in content repo.
+def install_all_content_packs_from_build_bucket(client: demisto_client, host: str, server_version: str,
+                                                bucket_packs_root_path: str, service_account: str,
+                                                extract_destination_path: str):
+    """ Iterates over the packs currently located in the Build bucket. Wrapper for install_packs.
+    Retrieving the metadata of the latest version of each pack from the index.zip of the build bucket.
 
     :param client: Demisto-py client to connect to the server.
     :param host: FQDN of the server.
     :param server_version: The version of the server the packs are installed on.
+    :param bucket_packs_root_path: The prefix to the root of packs in the bucket
+    :param service_account: Google Service Account
+    :param extract_destination_path: the full path of extract folder for the index.
     :return: None. Prints the response from the server in the build.
     """
     all_packs = []
-    logging.debug(f"Installing all content packs in server {host}")
+    logging.debug(f"Installing all content packs in server {host} from packs path {bucket_packs_root_path}")
 
-    for pack_id in os.listdir(PACKS_FULL_PATH):
-        if pack_id not in IGNORED_FILES:
-            metadata_path = os.path.join(PACKS_FULL_PATH, pack_id, PACK_METADATA_FILE)
-            with open(metadata_path, 'r') as json_file:
-                pack_metadata = json.load(json_file)
-                pack_version = pack_metadata.get('currentVersion')
-                server_min_version = pack_metadata.get('serverMinVersion', '6.0.0')
-                hidden = pack_metadata.get('hidden', False)
+    storage_client = init_storage_client(service_account)
+    build_bucket = storage_client.bucket(GCPConfig.CI_BUILD_BUCKET)
+    GCPConfig.STORAGE_BASE_PATH = bucket_packs_root_path  # Setting Storage base path
+    index_folder_path, _, _ = download_and_extract_index(build_bucket, extract_destination_path)
+
+    for pack_id in os.listdir(index_folder_path):
+        if os.path.isdir(os.path.join(index_folder_path, pack_id)):
```

Comment:
shouldn't you keep the `IGNORED_FILES` check?

Reply:
As we are not iterating over the repo, but over the index, we don't need this check. See https://github.com/demisto/content/blob/master/Tests/Marketplace/upload_packs.py#L45